pkg/webhook/preflight/nutanix/storagecontainer.go

```diff
@@ -0,0 +1,223 @@
+				nclient:  cd.nclient,
+			},
+		)
+	}
+
+	for mdName, nutanixWorkerNodeConfigSpec := range cd.nutanixWorkerNodeConfigSpecByMachineDeploymentName {
+		if nutanixWorkerNodeConfigSpec.Nutanix != nil {
+			checks = append(checks,
+				&storageContainerCheck{
+					nodeSpec: nutanixWorkerNodeConfigSpec.Nutanix,
+					field: fmt.Sprintf(
+						"cluster.spec.topology.workers.machineDeployments[.name=%s]"+
+							".variables[.name=workerConfig].value.nutanix",
+						mdName,
+					),
+					csiSpec: &cd.nutanixClusterConfigSpec.Addons.CSI.Providers.NutanixCSI,
+					nclient: cd.nclient,
+				},
+			)
+		}
+	}
+
+	return checks
+}
+
+func getStorageContainer(
+	client client,
+	nodeSpec *carenv1.NutanixNodeSpec,
+	storageContainerName string,
+) (*clustermgmtv4.StorageContainer, error) {
+	cluster, err := getCluster(client, &nodeSpec.MachineDetails.Cluster)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cluster: %w", err)
+	}
+
+	fltr := fmt.Sprintf("name eq '%s' and clusterExtId eq '%s'", storageContainerName, *cluster.ExtId)
+	resp, err := client.ListStorageContainers(nil, nil, &fltr, nil, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list storage containers: %w", err)
+	}
+
+	containers, ok := resp.GetData().([]clustermgmtv4.StorageContainer)
+	if !ok {
+		return nil, fmt.Errorf("failed to get data returned by ListStorageContainers(filter=%q)", fltr)
+	}
+
+	if len(containers) == 0 {
+		return nil, fmt.Errorf(
+			"no storage container named %q found on cluster named %q",
+			storageContainerName,
+			*cluster.Name,
+		)
+	}
+
+	if len(containers) > 1 {
+		return nil, fmt.Errorf(
+			"multiple storage containers found with name %q on cluster %q",
+			storageContainerName,
+			*cluster.Name,
+		)
+	}
+
+	return ptr.To(containers[0]), nil
+}
+
+func getCluster(
+	client client,
+	clusterIdentifier *v1beta1.NutanixResourceIdentifier,
+) (*clustermgmtv4.Cluster, error) {
+	switch {
+	case clusterIdentifier.IsUUID():
+		resp, err := client.GetClusterById(clusterIdentifier.UUID)
+		if err != nil {
+			return nil, err
+		}
+
+		cluster, ok := resp.GetData().(clustermgmtv4.Cluster)
+		if !ok {
+			return nil, fmt.Errorf("failed to get data returned by GetClusterById")
+		}
+
+		return &cluster, nil
+	case clusterIdentifier.IsName():
+		filter := fmt.Sprintf("name eq '%s'", *clusterIdentifier.Name)
+		resp, err := client.ListClusters(nil, nil, &filter, nil, nil, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		if resp == nil || resp.GetData() == nil {
+			return nil, fmt.Errorf("no clusters were returned")
+		}
+
+		clusters, ok := resp.GetData().([]clustermgmtv4.Cluster)
+		if !ok {
+			return nil, fmt.Errorf("failed to get data returned by ListClusters")
+		}
+
+		if len(clusters) == 0 {
+			return nil, fmt.Errorf("no clusters found with name %q", *clusterIdentifier.Name)
+		}
+
+		if len(clusters) > 1 {
+			return nil, fmt.Errorf("multiple clusters found with name %q", *clusterIdentifier.Name)
```

Comment:
At some point this probably deserves its own check

Reply:
Thanks! I've created a ticket to track this.